lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -747,14 +778,47 @@ fromShelleyTx (SL.Tx bod@(SL.TxBody ins outs certs wdrls _ _ _ _) _ mmd) =
     , mapMaybe fromShelleyDelegationCert (toList certs)
     , mapMaybe fromShelleyRegistrationCert (toList certs)
     )
+  where
+    SL.Tx bod@(SL.TxBody ins outs certs wdrls _ _ _ _) _ mmd = tx
+
+fromAllegraTx
+    :: ( SL.ShelleyBased era
+       , SL.Core.TxBody era ~ MA.TxBody era
+       , SL.Core.Value era ~ SL.Coin
+       , SL.Core.Metadata era ~ MA.Metadata era
+       , Ord (SL.Core.Script era)
+       )
```

Comment:
Perhaps there is a single AllegraEra type that satifies all of these constraints?